### ISSUE TITLE
Sensor gets hot and not working

### ISSUE THREAD

**Issue by lucacri** [brickbots/aball#4]
Hey! I got the board from Jack Enterprises - ADNS-9800 Laser Motion Sensor Breakout.

My problem is that i tried to use it in QMK with a RP2040. It worked really sporadically (like 3 times) when i had it on the breadboard and I jiggled the cables in the holes a bit. Once I saw that it worked by "jiggling", I thought that if I made the connections more permanent, it'd work all the time, but I was wrong.

After soldering (and re-do it twice again from scratch), the only thing that happens is that the ADN sensor gets hot to the touch (nothing crazy, not scalding but just pretty hot) and nothing works.

I'm at my wits end, do you have _any_ suggestion on how I could proceed?

**Comment by brickbots** [comment]
Yes, sounds like the 3.3 / 5v selectors might not be correct, or the jumpers have not been fully cut.  I've had a one where I thought I cut them well, but there was still continuity.  Definitely worth checking with a multimeter!

**Comment by lucacri** [comment]
I didn't touch the voltage selector since the RP2040 is using 3.3v. I even added some solder on the 3.3v pads just to make sure they were all touching. Tested with multimeter and it was 3.3v

I've detached it from the RP2040 for now but I really followed the schematic on the "electronics" page, and re-did the cabling twice just to make sure i didn't make a mistake...

**Comment by brickbots** [comment]
Okay, sounds like it's not a voltage select issue.  One item on the electronics diagram that could be clearer is that both the grounds needs to be bridged and connected back to the MCU.  The AG (Analog Ground) and DG (Digital Ground) are right next to each other on the sensor and need to be connected together.

It might also be worth using your phone camera to look at the sensor (try with and without the lens).  It uses IR light to illuminate the surface and most phones will show this.  It's a fair way to just see if the sensor is working and initialized.

**Comment by lucacri** [comment]
@brickbots you were definitely onto something! The two ground pins (14 and 11) were not properly connected to ground:

- PIN 14 <> GND = 0 ohms
- PIN 11 <> GND = 0.659k.

I attached a jumper cable between the two, and now the mouse is moving BUT:

- it starts by going straight to the top left of the screen
- trying to move my finger on top of the sensor does move the cursor, but extremely erratically and then it jumps back to the top left

I tried everything I could in QMK: changed the ADNS9800_CLOCK_SPEED, ADNS9800_SPI_DIVISOR, pins etc but nothing changes.

Any ideas?

_also thank you all for helping me!_

**Comment by brickbots** [comment]
Okay!  Getting closer.... The behavior you described reminded me of an issue another person had which they eventually solved, but not in a satisfying way.  Here's the relevant portion of the email exchange.  The behavior described sounds just like what is happening for you.  Diagonal motion to top-left of screen.

_I have an update regarding the behavior of my aball. 

I compile once again the firmware through qmk msys this time and flashed it with qmk toolbox. 

What I noticed this time ,(I don’t this has something to do with the way I compile the firmware) is that the aball works !!

But if I unplug the cable and plug it  again (same pc,same cable) then I get the behavior I desrcribed in the previous mails. 

The cursor moves on its own and sticks to the left end of my monitor. 
If I flash it again it works!!
If I restart my pc ,it’s still working._ 

Not sure what the lesson/take-away is from this.  I thought it might be something with the chip-select pin, but it seemed like it was soldered correctly, configured right in the source, and they did not change anything that I'm aware of to get it working.

**Comment by lucacri** [comment]
Hey gang, I SOLVED IT!

Remember when I said that I soldered and resoldered the cables twice? At the time the sensor could not work because of the AG and DG grounds not being attached, no matter what I did.

I just received a new great multimeter from Aliexpress, I wanted to test it so I just checked that the SPI pins were not shorting with each other, and.... the MISO and MOSI were shorted together!!! I desoldered one of them to see if it was my cable or the traces, and it was my cable because **ONE** single strand, literally the size of a human hair, was stuck to the other solder joint! 

I redid the soldering and SUCCESS! Works perfectly now!!

Thank you all for your help, without your suggestions (especially @brickbots with the ground pins idea) I would still be pulling my hair!
